tests/database.test.js

```diff
@@ -25,113 +25,164 @@ describe('Database Functionality', () => {
     });
 
     it('Should store user to database', async done => {
-        //Send request to register user
-        const res = await request.post('/api/register').send({
+        //Send request to register a user
+        var user = await request.post('/api/register').send({
             email: 'dummy@test.com',
             name: 'dummy test',
             password:'12345',
             is_professor: true
         });
 
         //Search user in database by email
-        const user = await User.findOne({
+        user = await User.findOne({
             email: 'dummy@test.com'
         });
 
-        //Check that user has name, email, password, and is or is not a professor
-        expect(user.name).toBeTruthy()
-        expect(user.email).toBeTruthy()
-        expect(user.password).toBeTruthy()
-        expect(typeof(user.is_professor) === 'boolean').toBeTruthy();
+        //Check that user information stored in the database is correct
+        expect(user.name).toBe('dummy test');
+        expect(user.email).toBe('dummy@test.com')
+        expect(user.password).toBe('12345')
+        expect(user.is_professor).toBe(true);
 
         //Delete test user after testing
-        response = await request.delete("/api/delete-user").send({
+        const response = await request.delete("/api/delete-user").send({
             email: 'dummy@test.com'
         });
+        expect(response.status).toBe(200);
 
         done ()
     });
 
     xit('Should store course to database', async done => {
 
+        //Create 2 students, an admin, and a professor
+        const student1 = await request.post('/api/register').send({
+            email: 'st1@test.com',
+            name: 'Student 1',
+            password:'st1234',
+            is_professor: false
+        });
+        const student2 = await request.post('/api/register').send({
+            email: 'st2@test.com',
+            name: 'Student 2',
+            password:'st2345',
+            is_professor: false
+        });
+        const admin1 = await request.post('/api/register').send({
+            email: 'admin@test.com',
+            name: 'An Admin',
+            password:'ad1234min',
+            is_professor: true
+        });
+        const prof1 = await request.post('/api/register').send({
+            email: 'prof1@test.com',
+            name: 'A Professor',
+            password:'pr1234of',
+            is_professor: true
+        });
```

Comment:
Every time you create something in the database (e.g. a user) in a test you need to delete it at the end. Otherwise, the database would get filled up with hundreds of them after running the tests multiple times (also specific to users: when you run the test again the users will already exist in the database so it will fail to create them since the email must be unique).

Reply:
Just added requests to delete the users at the end. Sorry, I thought I had done that already.